### DISCUSSION TITLE
Running other websites/services on same system as AdGuard Home?

### DISCUSSION THREAD

**Discussion by iroc409** [AdguardTeam/AdGuardHome#3680] [General]
Hello!

I am looking at switching from Pi-Hole to AdGuard Home because it is native to FreeBSD.  Right now I have a Proxmox server running containers, one with Pi-Hole and one with a small webpage for my home network and smokeping.  In the past, PiHole will pretty much wipe out anything else you're running on the system without doing some complicated stuff to make it work.  I'd like to change some hardware and software around, and I'd like to consolidate some hardware/software.

Can you readily run AdGuard Home on a system with other services, for example it runs in /AdGuard/ on the web server along with smokeping and a home page?  I tried looking around for an answer but couldn't find it.

Thanks!

**Comment by fernvenue** [comment]
> 
> 
> Hello!
> 
> I am looking at switching from Pi-Hole to AdGuard Home because it is native to FreeBSD. Right now I have a Proxmox server running containers, one with Pi-Hole and one with a small webpage for my home network and smokeping. In the past, PiHole will pretty much wipe out anything else you're running on the system without doing some complicated stuff to make it work. I'd like to change some hardware and software around, and I'd like to consolidate some hardware/software.
> 
> Can you readily run AdGuard Home on a system with other services, for example it runs in /AdGuard/ on the web server along with smokeping and a home page? I tried looking around for an answer but couldn't find it.
> 
> Thanks!

If you wanna provide all services on one port, I suggest you use [nginx](https://nginx.org) to proxy with different `server_name`. If you really only have one domain name available, maybe you can try [traefik](https://doc.traefik.io/traefik/routing/routers) path prefix, It allows your different services to work in different paths..

**Reply by iroc409** [reply]
Thanks fernvenue, that makes sense.  I'm hoping to do something a little less complex.  This is all internal for a home network, so there are no official domain names involved (just the local IP or system name).

What I'm hoping is that just AdGuard can run in its own folder on the web server... something like this:

Root folder/home page: http://server/
Smokeping: http://server/smokeping/
AdGuard: http://server/adguard/

Does that make sense?  This works fine with smokeping, and even works OK to an extent with PiHole, but once you run an update with PiHole it wipes anything else off the webserver.  There are some apparently complicated hacks to get around it, or you can try and run a script after, but it just doesn't play nice.

I don't think a proxy is necessary, but I'll look into that and traefik.  I've never messed with the latter at all.  Since this is a home network, I'm hoping to just make things as simple to maintain as possible.

Thank you again for the reply, it gives me more to look into.
